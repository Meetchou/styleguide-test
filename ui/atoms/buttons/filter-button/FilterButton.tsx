import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';

import Button, { ButtonProps } from '@ui/atoms/buttons/button/Button';

const FilterButton: FC<ButtonProps> = (props) => {
  const { t } = useTranslation();
  return (
    <Button
      icon='filterIcon'
      size='small'
      color='secondary'
      {...props}
    >
      {t('ui.common.filter')}
    </Button>
  );
};

export default FilterButton;
