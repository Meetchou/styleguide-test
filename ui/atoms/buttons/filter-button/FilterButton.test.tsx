import React from 'react';

import render from '@ui/testing/render';

import FilterButton from './FilterButton';

describe('atoms/button/FilterButton', () => {
  it('Render correctly', () => {
    const tree = render(<FilterButton />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
