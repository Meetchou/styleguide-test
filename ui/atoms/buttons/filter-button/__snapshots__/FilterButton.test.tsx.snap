// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/button/FilterButton Render correctly 1`] = `
<button
  className="relative text-center font-semibold transition-all text-sm rounded-lg border outline-none transition-colors flex items-center justify-center text-primary-500 bg-white border-gray-100 hover:bg-primary-100 hover:border-primary-100 h-8"
>
  <span
    className="relative mx-auto flex items-center gap-[10px] px-4"
  >
    <span>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24m0 24H0"
          fill="none"
        />
        <path
          d="M4.25 5.61C6.27 8.2 10 13 10 13v6c0 .55.45 1 1 1h2c.55 0 1-.45 1-1v-6s3.72-4.8 5.74-7.39A.998.998 0 0018.95 4H5.04c-.83 0-1.3.95-.79 1.61z"
        />
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
      </svg>
    </span>
    <span>
      Filtrer
    </span>
  </span>
</button>
`;
